crypto/ctype.c

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <e_os.h>
```

Comment:
OSSL_NELEM is not in e_os.h. e_os.h is meant to be used for really OS-specific things. Never include it unless you really have to.